retrier/retrier_test.go

```diff
@@ -153,6 +153,44 @@ func TestRetrierRunFnWithInfinite(t *testing.T) {
 	}
 }
 
+func TestRetrierCtxSurfaceWorkErrors(t *testing.T) {
```

Comment:
I know some of the other tests already duplicate across `Run`, `RunCtx`, and `RunFn` but given how `RunCtx` delegates to `RunFn` I think it’s probably sufficient to just test `RunFn`, and get rid of this test.

I would like to see a test for the inverse behaviour though (i.e. when the context is cancelled but `WithSurfaceWorkErrors` is not set) since I don’t believe any of the existing tests cover that case.